src/docs/04-java-client/18-workflow-replay-shadowing.md

```diff
@@ -0,0 +1,156 @@
+
+Here's a simple example. The example is also available [here](https://github.com/uber/cadence-java-samples/blob/master/src/test/java/com/uber/cadence/samples/hello/HelloWorkflowShadowingTest.java).
+
+```java
+public void testShadowing() throws Throwable {
+  IWorkflowService service = new WorkflowServiceTChannel(ClientOptions.defaultInstance());
+
+  ShadowingOptions options = ShadowingOptions
+          .newBuilder()
+          .setDomain(DOMAIN)
+          .setShadowMode(Mode.Normal)
+          .setWorkflowTypes(Lists.newArrayList("GreetingWorkflow::getGreeting"))
+          .setWorkflowStatuses(Lists.newArrayList(WorkflowStatus.OPEN, WorkflowStatus.CLOSED))
+          .setExitCondition(new ExitCondition().setExpirationIntervalInSeconds(60))
+          .build();
+  WorkflowShadower shadower = new WorkflowShadower(service, options, TASK_LIST);
+  shadower.registerWorkflowImplementationTypes(HelloActivity.GreetingWorkflowImpl.class);
+
+  shadower.run();
+}
+```
+
+### Shadowing Worker
+
+NOTE: 
+- **All shadow workflows are running in one Cadence system domain, and right now, every user domain can only have one shadow workflow at a time.**
+- **The Cadence server used for scanning and getting workflow history will also be the Cadence server for running your shadow workflow.** Currently, there's no way to specify different Cadence servers for hosting the shadowing workflow and scanning/fetching workflow.
+
+Your worker can also be configured to run in shadow mode to run shadow tests as a workflow. This is useful if there's a number of workflows that need to be replayed. Using a workflow can make sure the shadowing won't accidentally fail in the middle and the replay load can be distributed by deploying more shadow mode workers. It can also be incorporated into your deployment process to make sure there's no failed replay checks before deploying your change to production workers.
+
+When running in shadow mode, the normal decision worker will be disabled so that it won't update any production workflows. A special shadow activity worker will be started to execute activities for scanning and replaying workflows. The actual shadow workflow logic is controlled by Cadence server and your worker is only responsible for scanning and replaying workflows. 
+
+[Replay succeed, skipped and failed metrics](https://github.com/uber/cadence-java-client/blob/master/src/main/java/com/uber/cadence/internal/metrics/MetricsType.java#L169-L172) will be emitted by your worker when executing the shadow workflow and you can monitor those metrics to see if there's any incompatible changes. 
+
+To enable the shadow mode, the only change needed is setting the `EnableShadowWorker` field in `worker.Options` to `true`, and then specify the `ShadowOptions`. 
```

Comment:
Is it true for java client? For go client, only config needs to be changed. For java I thought there's a separate API for creating shadow worker?

Reply:
Yes. There is a separate class for it. I updated the doc here.